connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Cast.java

```diff
@@ -154,10 +158,14 @@ private R applyWithSchema(R record) {
         final Struct updatedValue = new Struct(updatedSchema);
         for (Field field : value.schema().fields()) {
             final Object origFieldValue = value.get(field);
+            log.trace("Cast SMT: Original value: {}", origFieldValue);
             final Schema.Type targetType = casts.get(field.name());
             final Object newFieldValue = targetType != null ? castValueToType(field.schema(), origFieldValue, targetType) : origFieldValue;
+            log.trace("Cast SMT: New value: {}", newFieldValue);
             updatedValue.put(updatedSchema.field(field.name()), newFieldValue);
         }
+        log.trace("Schema SMT application with updatedValue: {}, updatedSchema: {}",
+            updatedValue, updatedSchema);
```

Comment:
I'm not sure these add value since TransformationChain has log messages with the records, which should be sufficient to know whether the schemas are null.